aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -95,18 +97,18 @@ Tensor sparse_csr_tensor(
     c10::optional<bool> pin_memory) {
   // See [Note: hacky wrapper removal for TensorOptions]
   TensorOptions options = TensorOptions().dtype(dtype).layout(layout).device(device).pinned_memory(pin_memory);
-
   TORCH_CHECK(
       options.layout() == kSparseCsr,
       "expected sparse CSR layout, but got layout ",
       options.layout());
   TORCH_CHECK(crow_indices.numel() >= 1, "expected crow_indices.numel() >= 1, but got ",
               crow_indices.numel());
-  std::array<int64_t, 2> size;
 
+  std::array<int64_t, 2> size;
+  auto host_col_indices = crow_indices.is_cpu() ? col_indices : col_indices.to(kCPU);
```

Comment:
This looks a bit suspect. What's the invariant for when non-CPU is accepted or not?